### PR TITLE
Fix Deprecation::ignoreDeprecations

### PR DESCRIPTION
**Pull request by ruudk**
This applies @greg0ire's feedback and fixes #31

### CODE REVIEW

File: tests/Doctrine/Deprecations/DeprecationTest.php [hunk 1]
```diff
@@ -202,6 +202,23 @@ public function testDeprecationWithIgnoredPackage(): void
...
+
+        Deprecation::trigger(
+            'doctrine/orm',
+            'https://github.com/doctrine/orm/issue/1234',
+            'this is deprecated %s %d',
+            'foo',
+            1234
+        );
+
+        $this->assertEquals(1, Deprecation::getUniqueTriggeredDeprecationsCount());
```
[the remark this answers is not in the document]
Comment (ruudk) [reply tests/Doctrine/Deprecations/DeprecationTest.php]: I copied the test, but fixed it now. I think this makes more sense now indeed.

**Review by greg0ire — APPROVED** [review]
I think this makes more sense 👍

File: lib/Doctrine/Deprecations/Deprecation.php [hunk 2]
```diff
@@ -56,6 +56,9 @@ class Deprecation
     private static $ignoredPackages = [];
 
     /** @var array<string,int> */
+    private static $ignoredLinksCount = [];
```
Comment (stof) [inline lib/Doctrine/Deprecations/Deprecation.php:59]: the name looks weird to me here. Those counts will not be for *ignored* links (as ignored links will return before incrementing this)

Comment (ruudk) [reply lib/Doctrine/Deprecations/Deprecation.php]: Oef, you're right. Changed it now, is this better?

File: lib/Doctrine/Deprecations/Deprecation.php [hunk 3]
```diff
@@ -56,6 +56,9 @@ class Deprecation
     private static $ignoredPackages = [];
 
     /** @var array<string,int> */
+    private static $triggeredLinkCounts = [];
```
Comment (stof) [inline lib/Doctrine/Deprecations/Deprecation.php:59]: I suggest naming this property `$triggeredDeprecations`, like its getter

Comment (ruudk) [reply lib/Doctrine/Deprecations/Deprecation.php]: Even beter, changed.

**Review by stof — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @ruudk !

**Comment by ruudk** [comment]
Would it be possible to tag this? So that I can stop some deprecations on production? 😅

**Comment by greg0ire** [comment]
Sure thing. Out of interest, how urgent is it for you and why?

**Comment by ruudk** [comment]
Thanks for the tag! It was producing a 10-20 deprecations an hour. Great to be able to stop this 😃

**Comment by greg0ire** [comment]
Ok, so definitely not a money problem then I trust, just an output clutter issue? Cc @mpdude, I think this topic will interest you, w.r.t https://github.com/doctrine/doctrine-website/pull/485

**Comment by ruudk** [comment]
Deployed 1.1.1 and the deprecations stopped. Great 💙

**Pull request merged**